content/docs/rfcs/14/README.md

```diff
@@ -1,120 +1,102 @@
+## Message Attributes
 
-The `contentTopic` field SHOULD be filled out to allow for content-based filtering.
-See [12/WAKU2-FILTER](/spec/12) and [13/WAKU2-STORE](/spec/13) for more details.
-To enable a bidirectional bridge with Waku v1 see [15/WAKU2-BRIDGE](/spec/15) for further requirements on this field.
 
-The `version` field MAY be filled out to allow for various types of payload encryption.
-Omitting it means the version is 0.
+The `payload` attribute SHOULD contain the data to be sent.
 
-The `timestamp` field MAY be filled out to signify the time at which the message is generated by its sender. 
-This field holds the Unix epoch time in nanoseconds. 
-Omitting it means the timestamp is unspecified.
+The `contentTopic` attribute SHOULD specify a string identifier that can be used for content-based filtering.
 
-The `ephemeral` field MAY be set to signify the transient nature of the message.
-If the message should be stored by the [store protocol](/spec/13), then this field MUST be set to `false`, which is equivalent to omitting the field.
-If the message should not be stored by the [store protocol](/spec/13), then this field MUST be set to `true`.
+The `version` attribute MAY contain a version number to discriminate different types of payload encryption. If omitted, the value SHOULD be interpreted as version 0.
 
-See [13/WAKU2-STORE](/spec/13) for more details.
+The `timestamp` attribute signifies the time at which the message was generated by its sender. This field SHOULD contain the Unix epoch time in nanoseconds. If the attribute is omitted, it SHOULD be interpreted as timestamp 0.
 
-## Payloads
+The `ephemeral` attribute signifies the transient nature of the message. If this field is set to `true`, the message SHOULD be interpreted as ephemeral. Else, if the field is omitted or set to `false`, the message SHOULD be interpreted as non-ephemeral.
 
-Payloads are implemented using [protocol buffers v3](https://developers.google.com/protocol-buffers/).
+# Wire Format
+
+The `WakuMessage` wire format is specified using [protocol buffers v3](https://developers.google.com/protocol-buffers/).
 
 ```protobuf
 syntax = "proto3";
 
 message WakuMessage {
   bytes payload = 1;
-  string contentTopic = 2;
+  string content_topic = 2;
   uint32 version = 3;
   sint64 timestamp = 10;
   bool ephemeral = 31;
 }
 ```
 
-## Payload encryption
-
-Payload encryption depends on the `version` field.
-
-### Version 0
+# Payload encryption
 
-This indicates that the payload SHOULD be either unencrypted or that encryption is done at a separate layer outside of Waku.
+The waku message payload MAY be encrypted. The message `version` attribute indicates the schema used to encrypt the payload data.
 
-### Version 1
+- **Version 0:**  The payload SHOULD be interpreted as unencrypted; additionally, it CAN indicate that the message payload has been encrypted at the application layer.
```

Comment:
Why not `MUST`? i.e., `MUST` be interpreted?

Reply:
Same question for you, why not `SHOULD`? Can you elaborate?

Reply:
Yes, sure; for reference, based on https://rfc.vac.dev/spec/1/#language and the document referred therein https://www.rfc-editor.org/rfc/rfc2119, the term `SHOULD` indicates _recommendation_ whereas `MUST` indicates _an absolute requirement_. So in here, when the version is 0 (or 1), to me, the payload is absolutely unencrypted  (or encrypted), i.e., there is no doubt. Hence the term `MUST` looks more appropriate.

Reply:
Not sure I agree. When version is `0` it simply means that Waku is not enforcing some encryption scheme. Applications are free to encrypt the payload as they wish (and most do). `SHOULD` is therefore more appropriate, IMO.

Reply:
>  Applications are free to encrypt the payload as they wish (and most do). SHOULD is therefore more appropriate, IMO.

I see what you mean, right, for version 0 it make sense to use `SHOULD`. But in the other cases i.e., version value of 1 and 2, I think that needs to be updated to `MUST` following my previous reasoning.

Reply:
Or, are we going to argue that if the version is e.g., `1`, the application still can decide to use a different encryption scheme or none at all? @jm-clius if that is the case, then `SHOULD` is appropriate.